### ISSUE TITLE
Add general polygon

### ISSUE THREAD

**Issue by mforets** [JuliaReach/LazySets.jl#3115]
Add a struct that we can call `Polygon` that represents a possibly non-convex polygon. We can implement it as an array of line segments.

cc @mvanzulli

**Comment by schillic** [comment]
The line segments would store every vertex twice. A more efficient representation would be with only the vertices in the right order (`Plots` accepts the points both in clockwise and counter-clockwise order). A name could then be `VPolygonNC` ("NC" for "non-convex").

```julia
julia> vlist = [[0., 0], [0, 2], [2, 2], [2, 0], [1, 1]];
julia> vlistr = reverse(vlist);

# both orders of the vertices work
julia> plot([v[1] for v in vlist], [v[2] for v in vlist], seriestype=:shape)
julia> plot([v[1] for v in vlistr], [v[2] for v in vlistr], seriestype=:shape)
```

![tooth](https://user-images.githubusercontent.com/9656686/194691588-59e9619f-7a94-4acf-b72c-4d8f0bec0964.png)

**Comment by mvanzulli** [comment]
Good point, thanks Christian for the suggestion. Then we can add a method `line_segments(non_convex_poly::VPolygonNC)` that builds the vector of line segmentes. That would be desirable for my application in order to compute a ray_tracing method. I have included this comments in #3116.
